app/js/hoc/withFooter.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {Button} from 'reactstrap';
+
+import './withFooter.scss';
+
+const Footer = (props) => (
+  <footer className="container">
+    {props.wizard.canPrevious() &&
+      <Button onClick={props.previous} className="m-2" color="link">&lt; Previous</Button>}
+    {props.wizard.canNext() &&
+      <Button onClick={props.next} className="float-right m-2" color="link" disabled={!props.nextEnabled}>Next &gt;</Button>}
+  </footer>
+);
+
+Footer.propTypes = {
+  wizard: PropTypes.object,
+  nextEnabled: PropTypes.bool,
+  previous: PropTypes.func,
+  next: PropTypes.func
+};
+
+const withFooterHoC = (WrappedComponent, wizard) => {
+  class HOC extends Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        nextEnabled: false
+      };
+      this.changeSubs = [];
+    }
+
+    enableNext = () => {
+      this.setState({ nextEnabled: true });
+    };
+
+    disableNext = () => {
+      this.setState({ nextEnabled: false });
+    };
+
+    change() {
+      this.changeSubs.forEach(cb => {
+        cb.call(cb);
+      });
+      this.changeSubs = [];
```

Comment:
What does this one do?

Reply:
This is a very simple event fake bus. It's nice to only save the state to the redux state only on page change, so we register a callback with `onPageChange` and then, when the footer changes the page, it calls this.
We clear the array of callbacks after, because they should only be called once.

Reply:
Nice 👍

Reply:
Oh yes of course, I said **may be later**. I was just dropping idea here